File: Data Structures & Algorithms/two-integer-sum/submission-8.java
class Solution 
{
    public int[] twoSum(int[] nums, int target) 
    {
        Map<Integer, Integer> cache = new HashMap<Integer, Integer>();
        int[] res = new int[2];
        for(int i = 0; i < nums.length; i++)
        {
            int golden_num = target - nums[i];
            if (cache.containsKey(golden_num) )
            {
                res[0] = cache.get(golden_num);
                res[1] = i;

                return res;
            }
            cache.put(nums[i], i);
        }
        return res;
    }
}

File: Data Structures & Algorithms/binary-search/submission-8.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1

        while l <= r:
            half = (l + r) // 2
            
            if nums[half] == target: return half
            
            match target > nums[half]:
                case True:
                    l = half + 1
                case _:
                    r = half - 1
        
        return -1